### DISCUSSION TITLE
How to reconcile a HelmRelease if a value in valuesFile property changes?

### DISCUSSION THREAD

**Discussion by ghost** [fluxcd/flux2#965] [Q&A]
We have defined a HelmRelease with a valuesFile property referencing an override values file sitting in a flux git source. If I change a value in the valuesFile e. g. the image tag then the new version is only deployed, if I also increment the chart version. Is it somehow possible to also update my service if I change only a value in the valuesFile because the chart doesn't change so also the chart version shouldn't change. But I expect that the helm controller doesn't check for value changes in the values file on reconciliation. How should one handle updating the image tags without always increasing the chart version?

**Comment by hiddeco** [comment]
The `valuesFile` should only be used as a base layer, for what you want to do, you put them in the `valuesFrom` or `values` field of a  `HelmRelease`.

**Reply by ghost** [reply]
Ok, thanks for answering. Thought I could handle different env's with different valuesFiles and manage the image tag in the corresponding values file of the env but will try now to override the tag in the `HelmRelease`.

**Reply by hiddeco** [reply]
It may look like you can, but you can't, as the changes will not be detected without bumping the chart version.

**Reply by ghost** [reply]
You mean changing a value in the `values` section of the `HelmRelease` without bumping the chart version will not work if I do a reconcile on the `HelmRelease`?

**Comment by angelbarrera92** [comment]
Im in the same page, is there any workaround?

**Reply by o6uoq** [reply]
Did you bump the chart version on re-deploy?

**Comment by Cajga** [comment]
Seems we are hitting this as well. We changed some values in HelmRelease spec.values.* and cannot get it redeployed.
flux reconcile does not work and surprisingly not even flux delete helmrelease solved the problem.

I wonder what is the best way to notify flux to apply the changes in the spec.values of a HelmRelease.

**Reply by o6uoq** [reply]
Did you bump the chart version on re-deploy?

**Reply by Cajga** [reply]
So, it was on me, sorry for resurrecting this old ticket. Everything works as expected. Let me write here what has happened if someone runs into the same thing:
First of all, I could not bump the chart version as it is a third party helm chart and we use the latest version: Istiod. I only wanted to add an extra value into the spec.values in HelmRelease of Istiod. I just added it in a wrong format and helm/istio just ignored it so, it did not get into the Istiod pod:
```
  values:
    pilot.env:
        PILOT_ENABLE_ALPHA_GATEWAY_API: true
```
The correct format is:
```
  values:
    pilot:
      env:
        PILOT_ENABLE_ALPHA_GATEWAY_API: true
```

In fact, it is enough to modify the `spec.values` in a `HelmRelease` and after the commit, flux will reconcile and the new values get added (without the need to bump chart version as it is not relevant for a "config" change anyway). `helm get values -n flux-system istio-system-istiod` helped me to understand that flux indeed did the job as expected. Then I just had to find the root cause why it gets ignored :)
